PWGHF/HFC/TableProducer/correlatorDplusHadrons.cxx

```diff
@@ -119,168 +119,175 @@ struct HfCorrelatorDplusHadrons {
         }
-        entryDplusHadronPair(getDeltaPhi(track.phi(), candidate1.phi()),
-                             track.eta() - candidate1.eta(),
-                             candidate1.pt(),
-                             track.pt());
-        entryDplusHadronRecoInfo(invMassDplusToPiKPi(candidate1), (int)flagDplusSignal);
-      } // end inner loop (Tracks)
 
-    } // end outer Dplus loop
+        if (std::abs(candidate1.flagMcMatchRec()) == 1 << DecayType::DplusToPiKPi) {
+          // fill per-candidate distributions from Dplus true candidates
+          registry.fill(HIST("hPtCandMCRec"), candidate1.pt());
+          registry.fill(HIST("hPtProng0MCRec"), candidate1.ptProng0());
+          registry.fill(HIST("hPtProng1MCRec"), candidate1.ptProng1());
+          registry.fill(HIST("hPtProng2MCRec"), candidate1.ptProng2());
+          registry.fill(HIST("hEtaMCRec"), candidate1.eta());
+          registry.fill(HIST("hPhiMCRec"), RecoDecay::constrainAngle(candidate1.phi(), -o2::constants::math::PI / 2.));
+          registry.fill(HIST("hYMCRec"), yDplus(candidate1));
+          registry.fill(HIST("hSelectionStatusMCRec"), candidate1.isSelDplusToPiKPi());
+        }
+        // fill invariant mass plots from Dplus signal and background candidates
+        registry.fill(HIST("hMassDplusMCRec"), invMassDplusToPiKPi(candidate1), efficiencyWeight);
+        if (candidate1.flagMcMatchRec() == 1 << DecayType::DplusToPiKPi) { // also matched as Dplus
+          registry.fill(HIST("hMassDplusMCRecSig"), invMassDplusToPiKPi(candidate1), candidate1.pt(), efficiencyWeight);
+        } else {
+          registry.fill(HIST("hMassDplusMCRecBkg"), invMassDplusToPiKPi(candidate1), candidate1.pt(), efficiencyWeight);
+        }
+
+        // Dplus-Hadron correlation dedicated section
+        // if the candidate is selected as Dplus, search for Hadron and evaluate correlations
+        flagDplusSignal = candidate1.flagMcMatchRec() == 1 << DecayType::DplusToPiKPi;
+        for (const auto& track : tracks) {
+          if (std::abs(track.eta()) > etaTrackMax) {
+            continue;
+          }
+          if (track.pt() < ptTrackMin) {
+            continue;
+          }
+          // Removing Dplus daughters by checking track indices
+          if ((candidate1.prong0Id() == track.globalIndex()) || (candidate1.prong1Id() == track.globalIndex()) || (candidate1.prong2Id() == track.globalIndex())) {
+            continue;
+          }
+          if (std::abs(track.dcaXY()) >= dcaXYTrackMax || std::abs(track.dcaZ()) >= dcaZTrackMax) {
+            continue; // Remove secondary tracks
+          }
+          entryDplusHadronPair(RecoDecay::constrainAngle(getDeltaPhi(track.phi(), candidate1.phi()), -o2::constants::math::PI / 2.),
+                               track.eta() - candidate1.eta(),
+                               candidate1.pt(),
+                               track.pt());
```

Comment:
Also here you can remove the RecoDecay::constrainAngle call, since it's already contained within the getDeltaPhi method